fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/ProgressiveLoanTransactionValidatorImpl.java

```diff
@@ -149,6 +151,11 @@ public void validateCapitalizedIncome(final JsonCommand command, final Long loan
                 }
             }
 
+            final Long transactionClassificationId = fromApiJsonHelper
+                    .extractLongNamed(LoanApiConstants.TRANSACTION_CLASSIFICATIONID_PARAMNAME, element);
+            progressiveLoanTransactionService.validateClassificationCodeValue(LoanApiConstants.CAPITALIZED_INCOME_CLASSIFICATION_CODE,
```

Comment:
no need this to be in a standalone service, lets move the validation here please.